xcode/Subconscious/Shared/Services/DataService.swift

```diff
@@ -253,6 +253,12 @@ actor DataService {
         
         return results
     }
+    
+    func resetIndex() async throws {
+        let did = try await self.noosphere.identity()
+        try self.database.resetOurSphere(identity: did)
+        try self.database.resetPeers()
+    }
```

Comment:
Mentioned above, but how does this work with sphere->database synchronization?

We're pretty careful right now to keep database updates atomic and lock-step with sphere updates. Naively, to me, this would seem to violate those assumptions.

Reply:
As mentioned here: https://github.com/subconsciousnetwork/subconscious/pull/936#discussion_r1367990163 this is no different to it being the first time we index something. We're only resetting a pointer to force re-indexing from the beginning of history, not actually deleting any indexed data. I don't see how that can lead to a desync.